src/lib/InstantSearch.js

```diff
@@ -426,32 +424,29 @@ See: https://www.algolia.com/doc/guides/building-search-ui/widgets/create-your-o
   }
 }
 
-export function enhanceConfiguration(searchParametersFromUrl) {
-  return (configuration, widgetDefinition) => {
-    if (!widgetDefinition.getConfiguration) return configuration;
-
-    // Get the relevant partial configuration asked by the widget
-    const partialConfiguration = widgetDefinition.getConfiguration(
-      configuration,
-      searchParametersFromUrl
-    );
+export function enhanceConfiguration(configuration, widgetDefinition) {
+  if (!widgetDefinition.getConfiguration) {
+    return configuration;
+  }
 
-    const customizer = (a, b) => {
-      // always create a unified array for facets refinements
-      if (Array.isArray(a)) {
-        return union(a, b);
-      }
+  // Get the relevant partial configuration asked by the widget
+  const partialConfiguration = widgetDefinition.getConfiguration(configuration);
 
-      // avoid mutating objects
-      if (isPlainObject(a)) {
-        return mergeWith({}, a, b, customizer);
-      }
+  const customizer = (a, b) => {
+    // always create a unified array for facets refinements
+    if (Array.isArray(a)) {
+      return union(a, b);
+    }
 
-      return undefined;
-    };
+    // avoid mutating objects
+    if (isPlainObject(a)) {
+      return mergeWith({}, a, b, customizer);
```

Comment:
We already use Object.assign in other parts of the code right? Worth checking if we can use it here.

Reply:
I prefer to let this function as is (even though its implementation is not correct). This is a sensitive part of the code that I really don't want to alter at the moment. We'll get rid of the function with the federated search implementation anyway.